patroni/psycopg.py

```diff
@@ -14,27 +22,79 @@
     except ImportError:
         _legacy = True
 
-    def quote_literal(value, conn=None):
+    if TYPE_CHECKING:
+        from psycopg2.extensions import connection
+
+    def quote_literal(value: Any, conn: Optional["connection"] = None) -> str:
```

Comment:
We were on doubt about how to deal with that, I mean, if we should use `TYPE_CHECKING` when there is no issue with circular importing, but we are only importing `connection` because of type hinting. What are your thoughts on that?

Reply:
But it this case there are no circular dependencies, we already imported from `psycopg2.extensions` on the line 18.

Reply:
Yes, that we understood. What we were thinking was whether to make it clear what is imported _just_ for type checking and doesn't form part of the functional code. It seems that `TYPE_CHECKING` is mainly used to avoid circular dependencies and in some cases to avoid large imports for performance reasons, but it appears that, as a dev looking at the import list, that it might be advantageous to separate these use cases, as a documentation aid.

If that doesn't make sense, then yes this should be on line 18.